keyboard/src/modules/Keyboard.rb

```diff
@@ -116,6 +116,15 @@ def SetModified
       @modified = true
     end
 
+    # Reset all setting
+    def reset
```

Comment:
Where is this used? Please, document why it is needed.

Reply:
For the testcase only. Or do you see another chance to reset the current keyboard ?

Reply:
What about defining private attr_readers and then mock it?

~~~
private

attr_reader :curr_kbd
attr_reader :keyboard_on_entry
attr_reader :default_kbd
attr_reader :user_decision
attr_reader :modified

# In the test

allow(subject).to receive(:curr_kbd).and_return("")
~~~

Reply:
And make sure to use `!curr_kbd.empty?` instead of `!@curr_kbd.empty?` in the module code.